samples/nfc/tag_reader/src/main.c

```diff
@@ -467,76 +444,195 @@ static void t4t_isodep_data_send(u8_t *data, size_t data_len, u32_t ftd)
 
 	err = st25r3911b_nfca_transfer_with_crc(&tx_buf, &rx_buf, ftd);
 	if (err) {
-		printk("NFC T4T ISO-DEP transfer error: %d\n.", err);
+		printk("NFC T4T ISO-DEP transfer error: %d.\n", err);
 	}
 }
 
 static void t4t_isodep_received(const u8_t *data, size_t data_len)
 {
 	int err;
-	struct nfc_t4t_apdu_resp apdu_resp;
-
-	nfc_t4t_apdu_resp_clear(&apdu_resp);
 
-	err = nfc_t4t_apdu_resp_decode(&apdu_resp, data, data_len);
+	err = nfc_t4t_hl_procedure_on_data_received(data, data_len);
 	if (err) {
-		printk("NFC APDU data decode err %d.\n", err);
-		return;
-	}
-
-	if (apdu_resp.status == NFC_T4T_APDU_RAPDU_STATUS_CMD_COMPLETED) {
-		printk("NFC T4T app selected successfully.\n");
-	} else {
-		printk("NFC T4T app not selected.\n");
+		printk("NFC Type 4 Tag HL data received error: %d.\n", err);
 	}
+}
 
+static void t4t_isodep_deselected(void)
+{
 	st25r3911b_nfca_tag_sleep();
 
 	k_delayed_work_submit(&transmit_work, TRANSMIT_DELAY);
-
 }
 
 static const struct nfc_t4t_isodep_cb t4t_isodep_cb = {
 	.selected = t4t_isodep_selected,
+	.deselected = t4t_isodep_deselected,
 	.error = t4t_isodep_error,
 	.ready_to_send = t4t_isodep_data_send,
 	.data_received = t4t_isodep_received
 };
 
+static void t4t_hl_selected(enum nfc_t4t_hl_procedure_select type)
+{
+	int err = 0;
+
+	switch (type) {
+	case NFC_T4T_HL_PROCEDURE_NDEF_APP_SELECT:
+		printk("NFC T4T NDEF Application selected.\n");
+
+		err = nfc_t4t_hl_procedure_cc_select();
+		if (err) {
+			printk("NFC T4T Capability Container select error: %d.\n",
+				err);
+		}
+
+		break;
+
+	case NFC_T4T_HL_PROCEDURE_CC_SELECT:
+		printk("NFC T4T Capability Container file selected.\n");
+
+		err = nfc_t4t_hl_procedure_cc_read(&NFC_T4T_CC_DESC(t4t_cc));
+		if (err) {
+			printk("Capability Container read error: %d.\n", err);
+		}
+
+		break;
+
+	case NFC_T4T_HL_PROCEDURE_NDEF_FILE_SELECT:
+		printk("NFC T4T NDEF file selected.\n");
+
+		err = nfc_t4t_hl_procedure_ndef_read(&NFC_T4T_CC_DESC(t4t_cc),
+						     t4t.ndef[t4t.tlv_index], NFC_T4T_APDU_MAX_SIZE);
+		if (err) {
+			printk("NFC T4T NDEF file read error %d.\n", err);
+		}
+
+		break;
+
+	default:
+		break;
+	}
+
+	if (err) {
+		st25r3911b_nfca_tag_sleep();
+
+		k_delayed_work_submit(&transmit_work, TRANSMIT_DELAY);
+	}
+}
+
+static void t4t_hl_cc_read(struct nfc_t4t_cc_file *cc)
+{
+	int err;
+	struct nfc_t4t_tlv_block *tlv_block;
```

Comment:
both variables can go to a tighter scope

Reply:
Ok. Moved